reflex/components/core/upload.py

```diff
@@ -148,6 +169,8 @@ def create(cls, *children, **props) -> Component:
         Returns:
             The upload component.
         """
+        UploadComponentUsed.is_used = True
```

Comment:
instead of also setting this here, lets call `UploadFilesProvider.create()` down in `_get_app_wrap_components` and let the `UploadFilesProvider.create` set the flag.

Reply:
Like this?
```patch
diff --git a/reflex/components/core/upload.py b/reflex/components/core/upload.py
index 389e25a8..30856140 100644
--- a/reflex/components/core/upload.py
+++ b/reflex/components/core/upload.py
@@ -169,8 +169,6 @@ class Upload(Component):
         Returns:
             The upload component.
         """
-        UploadComponentUsed.is_used = True
-
         # get only upload component props
         supported_props = cls.get_props()
         upload_props = {
@@ -215,5 +213,5 @@ class Upload(Component):
     @staticmethod
     def _get_app_wrap_components() -> dict[tuple[int, str], Component]:
         return {
-            (5, "UploadFilesProvider"): UploadFilesProvider(),
+            (5, "UploadFilesProvider"): UploadFilesProvider.create(),
         }
```
With this patch the upload endpoint is always disabled for me

Reply:
fixed by moving the endpoint setup at the end of `compile_`